### PR TITLE
[IMP] website_slides: improve the error message on URL field

### PR DESCRIPTION
**Pull request by std-odoo**
Purpose
=======
- Parse the JSON returned by Google
- Show a custom message if the API key is invalid

Task #2082857

### CODE REVIEW

File: addons/website_slides/models/slide_slide.py [hunk 1]
```diff
@@ -745,6 +746,18 @@ def _parse_youtube_document(self, document_id, only_preview_fields):
...
+        try:
+            error = json.loads(error)
+            error = (error.get('error', {}).get('errors', []) or [{}])[0].get('reason')
+        except json.decoder.JSONDecodeError:
+            error = str(error)
+
+        if error == 'keyInvalid':
+            return _('Your Google API key is invalid, please update it into your settings.\nSettings > Website > Features > API Key')
+
+        return _('Could not fetch data from url. Document or access right not available:\n%s') % error
```
Comment (dbeguin) [inline addons/website_slides/models/slide_slide.py]:
```python
def _extract_google_error_message(self, result):
   """ Some docstring with code error structure example """
    try:
        error = json.loads(result).get('error', {}).get('errors', [{}])[0]
        reason = error.get('reason')
        if reason == 'keyInvalid':
            return _('Your Google API key is invalid, please update it into your settings.\nSettings > Website > Features > API Key')
    except json.decoder.JSONDecodeError:
        reason = str(result)

    return _('Could not fetch data from url. Document or access right not available:\n%s') % reason
```

File: addons/website_slides/controllers/main.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 import math
 
 from odoo import http, tools, _
+from odoo.tools import html_escape
```
Comment (dbeguin) [inline addons/website_slides/controllers/main.py]: you don't use it.

File: addons/website_slides/models/slide_slide.py (hunk 1 above)
Comment (std-odoo) [reply addons/website_slides/models/slide_slide.py]: @dbeguin there's no error structure returned ;) it's just a string, with the "error reason" returned by Google (if the structure is know). Else, the structure is unknow (happen if we forgot a case)

Comment (dbeguin) [reply addons/website_slides/models/slide_slide.py]: I was talking about Google response structure, not the one you return in the function.
Duh ! :smile:

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**